src/fileSyncer.ts

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as utilities from './utilities';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { SSHServer, FileCopyConfigs, FileCopyConfig } from './interfaces';
+import * as sshHelper from './sshRunner';
+import * as async from 'async';
+import { StatusBarAlignment } from 'vscode';
+
+const browseThePC = 'Browse the PC..';
+
+export enum TargetType {
+    cloudshell = 'cloudshell',
+    remotehost = 'remotehost'
+}
+
+
+export class FileSyncer {
+    protected _statusBar: vscode.StatusBarItem;
+    protected _outputChannel: vscode.OutputChannel;
+    protected _configuration: FileCopyConfigs;
+    protected _hosts: { key: string, string };
+
+    constructor(outputChannel: vscode.OutputChannel) {
+        this._outputChannel = outputChannel;
+        this._configuration = utilities.getCodeConfiguration('ansible', 'copyFileOnSave');
+        this._statusBar = vscode.window.createStatusBarItem(StatusBarAlignment.Right, 100);
+    }
+
+    public updateConfiguration(config: any): void {
+        // check if changed
+        let updatedConfig = this.hasConfigurationChanged(this._configuration, config);
+
+        this.copyFiles(updatedConfig);
+
+        this._configuration = config;
+    }
+
+    protected hasConfigurationChanged(oldConfig: FileCopyConfigs, newConfig: FileCopyConfigs) {
```

Comment:
This function name sounds like a function to detect whether there is change happens and the return type should be bool.

Reply:
fixed

Reply:
fixed